### PR TITLE
Added --input-type flag. #70

### PR DESCRIPTION
**Pull request by markbates**

### CODE REVIEW

File: cmd/format-coverage.go [hunk 1]
```diff
@@ -12,31 +14,54 @@ import (
...
+			for _, f := range formatterMap {
+				if _, err := f.Search(); err == nil {
+					formatOptions.In = f
+					break
+				}
+			}
+		}
+
+		if formatOptions.In == nil {
+			return errors.WithStack(errors.New("could not find any viable formatter"))
```
Comment (noahd1) [inline cmd/format-coverage.go]: Could you print out the available formatters here as well?

File: formatters/simplecov/simplecov.go [hunk 2]
```diff
@@ -4,18 +4,39 @@ import (
...
+func (f *Formatter) Search(paths ...string) (string, error) {
+	paths = append(paths, searchPaths...)
+	for _, p := range paths {
+		if _, err := os.Stat(p); err == nil {
+			f.Path = p
+			return p, nil
+		}
+	}
+
+	return "", errors.WithStack(errors.Errorf("could not find a valid file for the simplecov formatter. search paths: %s", strings.Join(paths, ", ")))
```
Comment (noahd1) [inline formatters/simplecov/simplecov.go]: This error message implies that we perhaps found one but it was invalid. Is this the case?

File: cmd/format-coverage.go (hunk 1 above)
Comment (markbates) [inline cmd/format-coverage.go]: Sure can.

File: cmd/format-coverage.go [hunk 3]
```diff
@@ -70,5 +95,6 @@ func (f CoverageFormatter) Save() error {
 func init() {
 	formatCoverageCmd.Flags().BoolVarP(&formatOptions.Print, "print", "p", false, "prints to standard out only")
 	formatCoverageCmd.Flags().StringVarP(&formatOptions.Output, "output", "o", ccDefaultCoveragePath, "output path")
+	formatCoverageCmd.Flags().StringVarP(&formatOptions.InputType, "input-type", "i", "", fmt.Sprintf("type of input source to use [%s]", strings.Join(formatterList, ", ")))
```
Comment (markbates) [inline cmd/format-coverage.go]: My bad. That was supposed to be `t`. Force of habit using the first initial.

File: formatters/simplecov/simplecov.go (hunk 2 above)
Comment (markbates) [inline formatters/simplecov/simplecov.go]: No. It's we couldn't find any files. Any better wording?

Comment (noahd1) [inline formatters/simplecov/simplecov.go]: Perhaps:

"Could not find any files in search paths for simplecov. Search paths were:"

**Comment by markbates** [comment]
@noahd1 changes made. please review. thanks.

**Comment by noahd1** [comment]
Think you missed addressing one small thing, probably not intentionally. See outstanding comment above.

**Comment by markbates** [comment]
@noahd1 fixed. missed it in the noise of the PR. sorry.

**Comment by ale7714** [comment]
@markbates sorry, but i'm reverting this. Our main app CI is broken for some reason is not using -`--input-type 'simplecov'` by default. (And we like living on the edge and we're using test-reporter head 😅 )

**Comment by markbates** [comment]
@ale7714 before you revert it, can you explain what the problem is? i'd rather fix than revert it.

**Pull request merged**